controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -583,3 +866,85 @@ func aggregateFromMachinesToKCP(input aggregateFromMachinesToKCPInput) {
 	// This last case should happen only if there are no provisioned machines, and thus without conditions.
 	// So there will be no condition at KCP level too.
 }
+
+type aggregateV1Beta2ConditionsFromMachinesToKCPInput struct {
+	controlPlane      *ControlPlane
+	machineConditions []string
+	kcpErrors         []string
+	condition         string
+	trueReason        string
+	unknownReason     string
+	falseReason       string
+	note              string
+}
+
+// aggregateV1Beta2ConditionsFromMachinesToKCP aggregates a group of conditions from machines to KCP.
+// Note: the aggregation is computed in way that is similar to how v1beta2conditions.NewAggregateCondition works, but in this case the
+// implementation is simpler/less flexible and it surfaces only issues & unknown conditions.
+func aggregateV1Beta2ConditionsFromMachinesToKCP(input aggregateV1Beta2ConditionsFromMachinesToKCPInput) {
+	// Aggregates machines for condition status.
+	// NB. A machine could be assigned to many groups, but only the group with the highest severity will be reported.
+	kcpMachinesWithErrors := sets.Set[string]{}
+	kcpMachinesWithUnknown := sets.Set[string]{}
+	kcpMachinesWithInfo := sets.Set[string]{}
+
+	for i := range input.controlPlane.Machines {
+		machine := input.controlPlane.Machines[i]
+		for _, condition := range input.machineConditions {
+			if machineCondition := v1beta2conditions.Get(machine, condition); machineCondition != nil {
+				switch machineCondition.Status {
+				case metav1.ConditionTrue:
+					kcpMachinesWithInfo.Insert(machine.Name)
+				case metav1.ConditionFalse:
+					kcpMachinesWithErrors.Insert(machine.Name)
+				case metav1.ConditionUnknown:
+					kcpMachinesWithUnknown.Insert(machine.Name)
+				}
+			}
+		}
+	}
+
+	// In case of at least one machine with errors or KCP level errors (nodes without machines), report false.
+	if len(input.kcpErrors) > 0 || len(kcpMachinesWithErrors) > 0 {
+		messages := input.kcpErrors
+		if len(kcpMachinesWithErrors) > 0 {
+			messages = append(messages, fmt.Sprintf("Following machines are reporting %s errors: %s", input.note, strings.Join(sets.List(kcpMachinesWithErrors), ", ")))
+		}
+		v1beta2conditions.Set(input.controlPlane.KCP, metav1.Condition{
+			Type:    input.condition,
+			Status:  metav1.ConditionFalse,
+			Reason:  input.falseReason,
+			Message: strings.Join(messages, ", "),
+		})
+		return
+	}
+
+	// Otherwise, if there is at least one machine with unknown, report unknown.
+	if len(kcpMachinesWithUnknown) > 0 {
+		v1beta2conditions.Set(input.controlPlane.KCP, metav1.Condition{
+			Type:    input.condition,
+			Status:  metav1.ConditionUnknown,
+			Reason:  input.unknownReason,
+			Message: fmt.Sprintf("Following machines are reporting %s unknown: %s", input.note, strings.Join(sets.List(kcpMachinesWithUnknown), ", ")),
```

Comment:
```suggestion
			Message: fmt.Sprintf("Following machines are reporting %s Unknown: %s", input.note, strings.Join(sets.List(kcpMachinesWithUnknown), ", ")),
```
Maybe, up to you

Reply:
I defer this to a follow up PR, I think we should make this aggregation logic to look like Aggregate in the new condition utils (but I want to keep this out of this PR)